Agents/ForecastingAgent/forecasting/utils/default_configs.py

```diff
@@ -42,45 +48,46 @@ def retrieve_settings():
                 logger.error(f'"{v}" is missing in environment variables.')
                 raise ValueError(f'"{v}" is missing in environment variables.')
             if globals()[v] == '':
-                logger.error(f'No "{v}" value has been provided in environment variables.')
-                raise ValueError(f'No "{v}" value has been provided in environment variables.')
+                warning_msg = f'No default "{v}" value has been provided in environment variables. '
+                warning_msg += f'Ensure that "{v}" is provided in HTTP request to the agent to avoid issues.'
+                logger.warning(warning_msg)
+
     else:
         # 2) Stack deployment: retrieve settings from Stack Clients
         logger.info('Deploying agent to stack "{STACK_NAME}".')
 
-        # Import stack_configs module only when needed to avoid import issues/
-        # potentially unnecessary installation of py4jps StackClients resource
-        from .stack_configs import retrieve_stack_settings
-
-        # Retrieve target Blazegraph name for data to instantiate
+        # Retrieve target Blazegraph namespace for data to instantiate
         NAMESPACE = os.getenv('NAMESPACE')
         if NAMESPACE is None:
             logger.error('"NAMESPACE" name is missing in environment variables.')
             raise ValueError('"NAMESPACE" name is missing in environment variables.')
         if NAMESPACE == '':
-            logger.error('No "NAMESPACE" value has been provided in environment variables.')
-            raise ValueError('No "NAMESPACE" value has been provided in environment variables.')
+            warning_msg = f'No default "NAMESPACE" value has been provided in environment variables. '
+            warning_msg += f'Ensure that "NAMESPACE" is provided in HTTP request to the agent to avoid issues.'
+            logger.warning(warning_msg)
 
         # Retrieve target PostgreSQL/PostGIS database name
         DATABASE = os.getenv('DATABASE')
         if DATABASE is None:
             logger.error('"DATABASE" name is missing in environment variables.')
             raise ValueError('"DATABASE" name is missing in environment variables.')
         if DATABASE == '':
-            logger.error('No "DATABASE" value has been provided in environment variables.')
-            raise ValueError('No "DATABASE" value has been provided in environment variables.')
-        if DATABASE != 'postgres':
+            warning_msg = f'No default "DATABASE" value has been provided in environment variables. '
+            warning_msg += f'Ensure that "DATABASE" is provided in HTTP request to the agent to avoid issues.'
+            logger.warning(warning_msg)
+        elif DATABASE != 'postgres':
             logger.warning(f'Provided "DATABASE" name {DATABASE} does not match default database name "postgres".')
         
         # Retrieve settings from Stack Clients
+        # TODO: Review and check what happens/is returned if empty strings are provided
```

Comment:
This part requires testing for stack deployment, i.e. What happens when no default values are provided? In case exceptions are thrown this would require rework

Reply:
> This part requires testing for stack deployment, i.e. What happens when no default values are provided? In case exceptions are thrown this would require rework

If no namespace environment variable is defined in the agent config json, then the container fails to start when attempting to do so with the stack manager. The container logs show `ERROR - "NAMESPACE" name is missing in environment variables`. I am happy with this behaviour, and the error message is sufficiently clear.